examples/openshift-provision/README.md

```diff
@@ -0,0 +1,131 @@
+  * At a minimum this must define the [`hierarchy`](#defining-hierarchy) for your cluster.
+* `cluster/<cluster_name>/base.yml`
+  * This file defines your OpenShift cluster and is passed to the
+    openshift-provision role as the variable
+    [`openshift_clusters`](https://github.com/gnuthought/ansible-role-openshift-provision#openshift_provision-or-openshift_clusters).
+
+#### Example vars structure
+
+```shell
+vars_repository
++-- cloud
+|   +-- gcpusc1
+|   |   +-- vars
+|   |   |   +-- main.yml
+|   +-- awsuse1
+|   |   +-- vars
+|   |   |   +-- main.yml
++-- cluster
+|   +-- my_cluster_name
+|   |   +-- base.yml
+|   |   +-- vars
+|   |   |   +-- main.yml
++-- default
+|   +-- vars
+|   |   +-- main.yml
++-- openshift_release
+|   +-- 3.11
+|   |   +-- vars
+|   |   |   +-- main.yml
+```
+
+### Resource repository
+
+#### Defining resources
+
+In each level of your hierarchy, you can provide resource definitions. These can be
+defined as yaml definitions, or as Jinja templates. All you need to do is use the
+`.j2` file extension of your resource file, and openshift-provision will process
+it as Jinja template before taking action on the resource.
+
+#### Pretasks and posttasks
+
+In the top level of the repository with your cluster resource definitions, if
+you include a `cluster-pretasks.yml` and/or `cluster-posttasks.yml` file that
+can be used to perform Ansible tasks to run before or after the running of the
+openshift-provision role. These should be in the format of a list of tasks,
+not a full playbook.
+
+#### Example resources structure
+
+```shell
+resource_repository
++-- cloud
+|   +-- gcpusc1
+|   |   +-- resources
+|   |   |   +-- main.yml
+|   +-- awsuse1
+|   |   +-- resources
+|   |   |   +-- main.yml
++-- cluster
+|   +-- my_cluster_name
+|   |   +-- resources
+|   |   |   +-- resource_a.yml
+|   |   |   +-- resource_b.yml.j2
++-- cluster-pretasks.yml
++-- cluster-posttasks.yml
++-- default
+|   +-- resources
+|   |   +-- resource_1.yml
+|   |   +-- resource_2.yml.j2
+```
+
+## Defining hierarchy
+
+The openshift-provision template processor allows you to define a hierarchy for
+your resources and variables. The definition of this hierarchy must be defined
+in `vars_repository/cluster/<cluster_name>/vars/main.yml` as a list variable
+named `hierarchy`.
+
+Any variable defined in `hierarchy` must have a corresponding folder at the paths
+`vars_repository/<key>/<value>/vars` and `resource_repository/<key>/<value>/resources`.
+When the template processor runs, it will retrieve all yaml and json files in the `vars`
+directory allowing you to organize your variables into logical groups.
+
+``` yaml
+$ cat vars_repository/cluster/my_cluster_name/vars/main.yml
+---
+hierarchy:
+- openshift_release: 3.11
+- cloud: gcpusc1
+- env: lle
+- cluster: my_cluster_name
+```
+
+In this example, the order of priority for variables and resource definitions
+for `my_cluster_name` would be
+
+1. cluster (Top priority)
+1. env
```

Comment:
In Markdown files, you can just do `1.` for each entry in the list and it will render with the proper numbering. That way if you change something in the middle you don't need to go through and re-number them.

Reply:
Yeah, I know; I just personally like to try and keep them readable and consistent as regular text files too :) But that's a nitpick-grade comment, so I don't insist and will accept your decision here.